src/main/java/com/joejoe2/chat/models/PrivateChannel.java

```diff
@@ -73,6 +81,29 @@ public User anotherMember(User member) {
     return getMembers().stream().filter(u -> !member.getId().equals(u.getId())).findFirst().get();
   }
 
+  public void setBlockage(User user, boolean isBlock) {
```

Comment:
Perhaps we can move all the business logic out of JPA entities to not violate SRP. As a rule of thumb JPA entities need to be as tiny as possible having no business logic inside.

Reply:
I don't think this is complicated enough to be an business logic. It's more like a setter with internal data transformation.

Reply:
Anyways this setter looks quite weighty, having setter - fine. Having setter with some if-else statement is not so good.

Reply:
Actually I have put every logic before I know the clean code or ddd. For now I usually try to put the simple/core/self related logic in my model. You can refer to https://martinfowler.com/bliki/AnemicDomainModel.html or other articles.

Reply:
An article states that this is an anti-pattern

Reply:
Which one ? The above article say that AnemicDomainModel (only getter/setter, without any logic) is an anti pattern.

Reply:
Go, go ahead like it is now. The thing is that across the industry we tend to avoid placing any kind of logic into the domain model and move it out as soon as at least one occurrence has been found.

It just leads to an additional coupling in the place where it's not needed. We have a service and DAO/repository layer to operate with the set state of this domain model.

Therefore domain model itself should be a plain, stateless object having nothing more than just getters/setters, equals & hashcode, toString perhaps builder pattern instead of setters if we'd like to introduce an immutable object. However, in the case of JPA domain models' the state could and should change, therefore we have setters.
But that state should change in service and DAO layer and not en entity itself.

Reply:
It can be left as it is now.

Usually, it's advised to avoid placing any kind of logic into the domain model and moving it out as soon as at least one occurrence has been found. Unless DDD is being applied.

It just leads to an additional coupling in places where it's not needed. The DAO/repository and service layers are usually on duty to change the state of a domain model.

Therefore domain model itself should be a plain, stateless object having nothing more than just getters/setters, equals & hashcode, toString perhaps builder pattern instead of setters if we'd like to introduce an immutable object. However, in the case of JPA domain models' (as per JSR specification) state could and should change, therefore we have setters.
But that state should change in service and respectively DAO layer and not an entity itself.

However, if DDD is being followed then it allows having some sort of business logic inside of the domain model.

So, it depends on which architectural approach you'd like to choose :)

But since this project itself is not fully DDD I have doubts here if...